src/components/Common/ReorderableList/DropZone.tsx

```diff
@@ -0,0 +1,128 @@
+import { useRef, useEffect, memo } from 'react'
+import classnames from 'classnames'
+import { useDrop } from 'react-dnd'
+import styles from './ReorderableList.module.scss'
+import { ITEM_TYPE } from './constants'
+
+interface DropZoneProps {
+  position: number
+  isActive: boolean
+  onActivate: () => void
+  onDeactivate: () => void
+  onDrop: (fromPosition: number) => void
+  listId: string
+  className?: string
+}
+
+export const DropZone = memo(function DropZone({
+  position,
+  isActive,
+  onActivate,
+  onDeactivate,
+  onDrop,
+  listId,
+  className,
+}: DropZoneProps) {
+  const ref = useRef<HTMLDivElement>(null)
+
+  // Track if the drop operation has been handled
+  const dropHandledRef = useRef(false)
+
+  // Add a debounce timer for hover state changes to prevent flickering
+  const debounceTimerRef = useRef<ReturnType<typeof setTimeout> | null>(null)
+  const isHoveringRef = useRef(false)
+
+  const [{ isOver }, drop] = useDrop(
+    () => ({
+      accept: ITEM_TYPE,
+      hover: (item: { index: number; listId: string }, monitor) => {
+        if (item.listId !== listId) return
+
+        const clientOffset = monitor.getClientOffset()
+        if (!clientOffset || !ref.current) return
+
+        if (isHoveringRef.current) return
+
+        isHoveringRef.current = true
+
+        if (debounceTimerRef.current) {
+          clearTimeout(debounceTimerRef.current)
+        }
+
+        onActivate()
+      },
+      collect: monitor => ({
+        isOver: !!monitor.isOver(),
+      }),
+      drop: (item: { index: number; listId: string }) => {
+        if (item.listId !== listId) return
+
+        if (dropHandledRef.current) return
+        dropHandledRef.current = true
+
+        setTimeout(() => {
+          dropHandledRef.current = false
+        }, 100)
```

Comment:
Do we need to consider clearing this timeout?

Reply:
Fixed